.github/workflows/build-and-test.yml

```diff
@@ -58,9 +58,12 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        clickhouse: [ 22.3, 22.8, 23.3, 23.8 ]
+        clickhouse: [ 23.8, 24.3, 24.4, 24.5, cloud]
```

Comment:
this silently removes testing for 22.3 22.8 23.3, I know that ClickHouse Inc is aggressive in removing the support for old versions, but AFAIK, there are still lots of users running older versions in production.

I would suggest keeping the older version unless it introduces significant maintenance efforts.

such change should happen in dedicated PR, instead of silent changes.

Reply:
I opened a dedicated PR to handle the ClickHouse Java client upgrading https://github.com/ClickHouse/spark-clickhouse-connector/pull/342